spring-pulsar/src/main/java/org/springframework/pulsar/listener/PulsarContainerProperties.java

```diff
@@ -246,4 +257,35 @@ public void setPulsarConsumerProperties(Properties pulsarConsumerProperties) {
 		this.pulsarConsumerProperties = pulsarConsumerProperties;
 	}
 
+	@SuppressWarnings("unchecked")
+	public void updateContainerProperties() {
+		if (!this.pulsarConsumerProperties.isEmpty()) {
+			if (this.pulsarConsumerProperties.containsKey(SUBSCRIPTION_NAME)) {
```

Comment:
One way to reduce this boiler plate would be w/ something like this...
```java
applyPropIfSpecified(SUBSCRIPTION_NAME, this::setSubscriptionName);
```

```java
	private <T> void applyPropIfSpecified(String key, Consumer<T> setter) {
		if (this.pulsarConsumerProperties.containsKey(key)) {
			T value = (T) this.pulsarConsumerProperties.get(key);
			setter.accept(value);
		}
	}
```
Wdyt?